src/realm/link_view.cpp

```diff
@@ -310,13 +310,18 @@ void LinkView::sort(size_t column_index, bool ascending)
 }
 
 
-void LinkView::sort(const SortDescriptor& order)
+void LinkView::sort(SortDescriptor&& order)
 {
+    if (REALM_UNLIKELY(!order)) {
+        throw LogicError(LogicError::empty_descriptor);
+    }
     if (Replication* repl = get_repl()) {
         // todo, write to the replication log that we're doing a sort
```

Comment:
It appears that would be an optimisation to just send a "sort" instruction rather than sending all rows in a different order.